lzokay-c/lzokay-c.cpp

```diff
@@ -0,0 +1,14 @@
+#include "lzokay-c.h"
+#include "../lzokay.hpp"
+
+extern "C"
+int lzokay_decompress(const uint8_t * src, size_t src_size, uint8_t *output, size_t *output_len)
+{
+    size_t needed_size = 0;
+    lzokay::EResult error =
+        lzokay::decompress(src, src_size, output, *output_len, needed_size);
+    if (error < lzokay::EResult::Success)
+        return LZOKAY_ERROR;

```

Comment:
I think it would be useful to re-define all error codes for the C API

Reply:
OK, I updated the patchset.

I also ended rewriting the lib in C as I needed it to build on more platform. The [resulting code is here](https://github.com/Matroska-Org/foundation-source/tree/coremake-out/8/libmatroska2/lzokay). I tested the decompression against minilzo compressed content and it worked perfectly.